docs/se/grpc/server.adoc

```diff
@@ -0,0 +1,933 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019, 2022 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= gRPC Server
+:description: Helidon gRPC Server
+:keywords: helidon, grpc, java, se
+:feature-name: gRPC Server
+:rootdir: {docdir}/../..
+
+include::{rootdir}/includes/se.adoc[]
+
+== Contents
+
+- <<Overview, Overview>>
+- <<Maven Coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<Configuration, Configuration>>
+- <<Examples, Examples>>
+
+== Overview
+Helidon gRPC Server provides a framework for creating link:http://grpc.io/[gRPC] applications.
+While it allows you to deploy any standard gRPC service that
+implements `io.grpc.BindableService` interface, including services generated
+from the Protobuf IDL files (and even allows you to customize them to a certain
+extent), using Helidon gRPC framework to implement your services has a number of
+benefits:
+
+* It allows you to define both HTTP and gRPC services using similar programming
+model, simplifying learning curve for developers.
+
+* It provides a number of helper methods that make service implementation
+significantly simpler.
+
+* It allows you to configure some of the Helidon value-added features, such
+as <<Security, security>> and <<Service Metrics, metrics collection>>
+down to the method level.
+
+* It allows you to easily specify custom marshaller for requests and
+responses if Protobuf does not satisfy your needs.
+
+* It provides built in support for <<Service Health Checks, health checks>>.
+
+
+=== Experimental
+
+WARNING: The Helidon gRPC feature is currently experimental and the APIs are
```

Comment:
Use Caution to indicate the possibility of damage to a program, system, or data. The caution notice should always precede the step where the risk occurs. Warning indicates potential bodily harm.

Reply:
@ljamen, follow-up changes where made in another PR: https://github.com/oracle/helidon/pull/4606 that removes the experimental warning.

Essentially, the whole gRPC doc changed is composed of this PR and https://github.com/oracle/helidon/pull/4606.